pkg/utils/nodes.go

```diff
@@ -33,3 +34,29 @@ func IsNodeNameValid(r client.Reader, nodeName string) (bool, error) {
 	}
 	return true, nil
 }
+
+// used for making new node object for test and have a unique resourceVersion
+// GetNode returns a node object with the name nodeName based on the nodeType input
+func GetNode(nodeType, nodeName string) *corev1.Node {
```

Comment:
IMHO nodeRole wold be a better name than nodeType?

Reply:
I agree